cpp/src/arrow/compute/api_vector.h

```diff
@@ -177,21 +174,17 @@ class ARROW_EXPORT RankOptions : public FunctionOptions {
   };
 
   explicit RankOptions(std::vector<SortKey> sort_keys = {},
-                       NullPlacement null_placement = NullPlacement::AtEnd,
                        Tiebreaker tiebreaker = RankOptions::First);
   /// Convenience constructor for array inputs
   explicit RankOptions(SortOrder order,
-                       NullPlacement null_placement = NullPlacement::AtEnd,
```

Comment:
I think that we should keep this signature to specify `NullPlacement`.

Reply:
The logic is to retain this signature and then pass it into SortKey, right?

Reply:
Sorry, I can't understand what you said...

I think that we need this change:

```diff
diff --git a/cpp/src/arrow/compute/api_vector.h b/cpp/src/arrow/compute/api_vector.h
index 0233090ef..64d6c4112 100644
--- a/cpp/src/arrow/compute/api_vector.h
+++ b/cpp/src/arrow/compute/api_vector.h
@@ -183,7 +183,7 @@ class ARROW_EXPORT RankOptions : public FunctionOptions {
   explicit RankOptions(SortOrder order,
                        NullPlacement null_placement = NullPlacement::AtEnd,
                        Tiebreaker tiebreaker = RankOptions::First)
-      : RankOptions({SortKey("", order)}, null_placement, tiebreaker) {}
+      : RankOptions({SortKey("", order, null_placement)}, tiebreaker) {}
 
   static constexpr char const kTypeName[] = "RankOptions";
   static RankOptions Defaults() { return RankOptions(); }
```

Do you want to say that we don't need to specify `NullPlacement` here?

Reply:
It really shouldn’t be deleted, this interface has been rolled back